api/envoy/extensions/http/header_validators/envoy_default/v3/header_validator.proto

```diff
@@ -0,0 +1,112 @@
+      UNESCAPE_AND_REDIRECT = 3;
+
+      // Unescape %2F and %5C sequences.
+      // Note: this option should not be enabled if intermediaries perform path based access control as
+      // it may lead to path confusion vulnerabilities.
+      UNESCAPE_AND_FORWARD = 4;
+    }
+
+    // Should paths be normalized according to RFC 3986?
+    // This operation overwrites the original request URI path and the new path is used for processing of
+    // the request by HTTP filters and proxied to the upstream service.
+    // Envoy will respond with 400 to requests with malformed paths that fail path normalization.
+    // The default behavior is to normalize the path.
+    // This value may be overridden by the runtime variable
+    // :ref:`http_connection_manager.normalize_path<config_http_conn_man_runtime_normalize_path>`.
+    // See `Normalization and Comparison <https://datatracker.ietf.org/doc/html/rfc3986#section-6>`_
+    // for details of normalization.
+    // Note that Envoy does not perform
+    // `case normalization <https://datatracker.ietf.org/doc/html/rfc3986#section-6.2.2.1>`_
+    bool skip_path_normalization = 1;
+
+    // Determines if adjacent slashes in the path are merged into one.
+    // This operation overwrites the original request URI path and the new path is used for processing of
+    // the request by HTTP filters and proxied to the upstream service.
+    // Setting this option to true will cause incoming requests with path `//dir///file` to not match against
+    // route with `prefix` match set to `/dir`. Defaults to `false`. Note that slash merging is not part of
+    // `HTTP spec <https://datatracker.ietf.org/doc/html/rfc3986>`_ and is provided for convenience.
+    bool skip_merging_slashes = 2;
+
+    // The action to take when request URL path contains escaped slash sequences (%2F, %2f, %5C and %5c).
+    // This operation may overwrite the original request URI path and the new path is used for processing of
+    // the request by HTTP filters and proxied to the upstream service.
+    PathWithEscapedSlashesAction path_with_escaped_slashes_action = 3;
+  }
+
+  message Http1ProtocolOptions {
+    // Allows Envoy to process HTTP/1 requests/responses with both `Content-Length` and `Transfer-Encoding`
+    // headers set. By default such messages are rejected, but if option is enabled - Envoy will
+    // remove the Content-Length header and process the message.
+    // See `RFC7230, sec. 3.3.3 <https://datatracker.ietf.org/doc/html/rfc7230#section-3.3.3>`_ for details.
+    //
+    // .. attention::
+    //   Enabling this option might lead to request smuggling vulnerabilities, especially if traffic
+    //   is proxied via multiple layers of proxies.
+    bool allow_chunked_length = 1;
+  }
+
+  Http1ProtocolOptions http1_protocol_options = 1;
+
+  // URI path normalization options.
+  // URI path is left unmodified if this configuration was not specified (the default). Note that omitting path
```

Comment:
The URI path
also I'd be inclined to swap - the URI path will be ... if config was not specified
or the URL path is ....  if the configuration is not specified

Reply:
Done